### PR TITLE
Updated DocumentReference R4 Resource

### PR DESCRIPTION
**Pull request by kckeim**
Tons of formatting updates in this one, and substantial re-wording of sections to be more descriptive and in a similar style to other resources that were recently updated.
Verified good links through document.
Validated that API calls behave as expected with both the expected and unexpected parameters.

Description
----
- Backtick'd a lot of elements throughout the entire page, including some linked elements
- Overview paragraph updated
- Links to FHIR-controlled fields updated
  - Custodian link wasn't pointing to r4 specifically - adjusted
  - Per PR#349, the Document description/title link is no longer supported and needed removed
    - https://github.com/cerner/fhir.cerner.com/pull/349
- Added links out to the R4 Binary resource in various places, since that resource gets referred to by this one.
- Search's Parameter table updated, and elements were re-orderd to put `date` and `period` next to each other
- Search's Implementation notes greatly expanded, based on API behavior during testing
  - `date` and `period` specifically needed adjustments on behavior, as they were substantially lacking in detail
  - Per PR#959, the notes about `period` specifically were not consistent with the API's behavior
    - https://github.com/cerner/fhir.cerner.com/pull/959
  - Added detail about the `_id`, and `encounter` behaviors
  - Adjusted the `_revinclude` details to mimic more recently documentation updates
- Implementation Notes for Retrieve, Create, Update, updated also, though less extensively
- $docref operation parameters and Implementation Notes were updated
- Anonymized the x-ref IDs for the Create and Updated responses
- Fixed Typo in the `R4_DOCUMENT_REFERENCE` example in the RUBY file - had a bonus double quote in there.
- YAML file corrections
  - consistency around MIME vs. mime
  - needed to escape the hyphens in the YAML file - they were breaking the code blocking from the backticks, so we got to use HTML tags instead
  - fixed some "angled" doublequotes
  - ensured we were aiming to R4-specific resources for many links that weren't already

*****************************
PR Checklist
----
- [X] Screenshot(s) of changes attached before changes merged.
- [X] Screenshot(s) of changes attached after changes merged and published.

*****************************
Old Overview:
image

New Overview:
image

*****************************
Old Parameters table:
image

New Parameters table:
image

*****************************
Old Search Implementation Notes:
image

New Search Implementation Notes: 
image
image

*****************************
Old Retrieve Implementation Notes:
image

New Retrieve Implementation Notes: 
image

*****************************
Old Create Implementation Notes:
image

New Create Implementation Notes: 
image

*****************************
Old Update Implementation Notes:
image

New Update Implementation Notes: 
image

*****************************
$docref old parameters and notes:
image

$docref new parameters and notes:
image

### CODE REVIEW

**Review by RicardoCrespo-ora — APPROVED** [review]

**Review by napster235 — APPROVED** [review]

File: content/millennium/r4/foundation/documents/document-reference.md [hunk 1]
```diff
@@ -37,9 +39,15 @@ The following fields are returned if valued for clinical documents:
 * [Patient encounter]( https://hl7.org/fhir/r4/documentreference-definitions.html#DocumentReference.context.encounter){:target="_blank"}
   * [Reference](https://hl7.org/fhir/r4/references.html#Reference){:target="_blank"} ([Encounter](https://hl7.org/fhir/r4/encounter.html){:target="_blank"})
 * [Document period]( https://hl7.org/fhir/r4/documentreference-definitions.html#DocumentReference.context.period){:target="_blank"}
-* [Custodian]( https://www.hl7.org/fhir/documentreference-definitions.html#DocumentReference.custodian){:target="_blank"}
+* [Custodian]( https://www.hl7.org/fhir/r4/documentreference-definitions.html#DocumentReference.custodian){:target="_blank"}
 * [Related](https://hl7.org/fhir/r4/documentreference-definitions.html#DocumentReference.context.related){:target="_blank"}
```
Comment (amcginn) [inline content/millennium/r4/foundation/documents/document-reference.md:43]: nit: can you make this section consistent and remove the spaces before the URLs in the links

Comment (kckeim) [reply content/millennium/r4/foundation/documents/document-reference.md:43]: lol, yes

File: content/millennium/r4/foundation/documents/document-reference.md [hunk 2]
```diff
@@ -246,7 +241,7 @@ Date: Fri, 04 Nov 2022 10:15:11 GMT
 Etag: W/"1"
 Location: https://fhir-ehr-code.cerner.com/r4/ec2458f2-1e24-41c8-b71b-0e701af7583d/DocumentReference/201051629
 Vary: Origin
-X-Request-Id: 5f5324e1-2966-47ec-8076-0d3906828b56+5532_p0Ds
+X-Request-Id: 1111111111111111111111111111111111111111111111
```
Comment (amcginn) [inline content/millennium/r4/foundation/documents/document-reference.md:244]: use a UUID

File: content/millennium/r4/foundation/documents/document-reference.md [hunk 3]
```diff
@@ -246,7 +241,7 @@ Date: Fri, 04 Nov 2022 10:15:11 GMT
...
 </pre>
```
Comment (amcginn) [inline content/millennium/r4/foundation/documents/document-reference.md:245]: add the `opc-request-id`

File: content/millennium/r4/foundation/documents/document-reference.md [hunk 4]
```diff
@@ -295,18 +294,12 @@ Date: Fri, 04 Nov 2022 11:54:04 GMT
 Etag: W/"2"
 Last-Modified: Fri, 04 Nov 2022 11:54:04 GMT
 Vary: Origin
-X-Request-Id: d2a211a4-ccd4-4ba8-b4f4-d3da9200543c+6313_U7Z3
+X-Request-Id: 1111111111111111111111111111111111111111111111
```
Comment (amcginn) [inline content/millennium/r4/foundation/documents/document-reference.md:297]: UUID

File: lib/resources/r4/document_reference.yaml [hunk 5]
```diff
@@ -440,9 +440,9 @@ fields:
   type: Code
   action:
     - update
-  description: Identifies the type of the data in the attachment and allows a method to be chosen to interpret or render the data. Includes mime type parameters such as charset where appropriate.
+  description: Identifies the type of the data in the attachment and allows a method to be chosen to interpret or render the data. Includes MIME type parameters such as charset where appropriate.
```
Comment (amcginn) [inline lib/resources/r4/document_reference.yaml:443]: thoughts on code for `charset`?

**Review by amcginn — APPROVED** [review]

File: lib/resources/r4/document_reference.yaml (hunk 5 above)
Comment (kckeim) [reply lib/resources/r4/document_reference.yaml:443]: Not opposed, but we don't currently code that throughout all the .yaml files in the repository.
it's only backtick'd when it's part of a longer phrase.
e.g. 
>   note: Must be provided. Content Type must include a supported MIME type and character set. The supported values are `application/pdf`, `text/plain;charset=utf-8`, `text/richtext;charset=utf-8`, `text/rtf;charset=utf-8`, `text/html;charset=utf-8`, `application/xml;charset=utf-8`, and `application/xhtml+xml;charset=utf-8`

Sourced in this same document:
https://github.com/cerner/fhir.cerner.com/blob/b1d03f7c0a09863de603dcff2f68ed0d612af975/lib/resources/r4/document_reference.yaml#L200

File: content/millennium/r4/foundation/documents/document-reference.md (hunk 2 above)
Comment (kckeim) [reply content/millennium/r4/foundation/documents/document-reference.md:244]: yessir

File: content/millennium/r4/foundation/documents/document-reference.md (hunk 3 above)
Comment (kckeim) [reply content/millennium/r4/foundation/documents/document-reference.md:245]: done

File: content/millennium/r4/foundation/documents/document-reference.md (hunk 4 above)
Comment (kckeim) [reply content/millennium/r4/foundation/documents/document-reference.md:297]: done

**Comment by Someshnsn** [comment]
After Deployment Screenshot:
Screenshot 2023-11-11 at 10 19 12 PM
Screenshot 2023-11-11 at 10 19 55 PM
Screenshot 2023-11-11 at 10 20 45 PM
Screenshot 2023-11-11 at 10 22 23 PM
Screenshot 2023-11-11 at 10 22 58 PM
Screenshot 2023-11-11 at 10 23 29 PM
Screenshot 2023-11-11 at 10 24 38 PM

**Pull request merged**